kratos/python_scripts/analysis_stage.py

```diff
@@ -124,12 +124,22 @@ def InitializeSolutionStep(self):
         """This function performs all the required operations that should be executed
         (for each step) BEFORE solving the solution step.
         """
+        # We reinitialize if remeshed previously
+        if self._CheckIfModelIsModified():
+            self.ClearDatabase()
+            self.ReInitializeSolver()
+
         self.PrintAnalysisStageProgressInformation()
 
         self.ApplyBoundaryConditions() #here the processes are called
         self.ChangeMaterialProperties() #this is normally empty
         self._GetSolver().InitializeSolutionStep()
 
+        # We reinitialize if remeshed on the InitializeSolutionStep
+        if self._CheckIfModelIsModified():
```

Comment:
so remeshing can happen in solver's `InitializeSolutionStep` or somewhere else .... 
This is where the ambiguity lies. We have to define a specific place where remeshing/modification happens.

Reply:
Before InitializeSolutionStep

Reply:
> depends of the strategy

which strategy ?

Reply:
I mean strategy as the procedure of remeshing followed, which may vary depending of the application

Reply:
Exactly, if the re-meshing depends on different strategies (in the sense you said above)  this reinitialize and clear database should be called in the overwritten functions of that particular file and not in the base analysis.

Reply:
This would mean different analysis for different problems, what is quite inconvenient, when adding  two lines in the base solves that issue. And these lines does not conflict between them

Reply:
Additionally that will be needed also in case we finally implement an adaptive NR (in fact this design was agreed time ago because of that reason)

Reply:
What if in future we come up with a strategy where remeshing is done in a place than that of the above two possibilities (hypothetical) then shall we also add that check/if/function  to the base ?

Reply:
I don't think there are more places where to execute remesh

Reply:
> This would mean different analysis for different problems, what is quite inconvenient, when adding two lines in the base solves that issue. And these lines does not conflict between them

I think this is exactly how these classes are intended to be used. There is a structural `static` and `dynamic` analysis no ? Similarly there could `RemeshingAnalysisTypeQ` and `RemeshingAnalysisTypeP`.